src/turn.c

```diff
@@ -127,8 +127,12 @@ static int find_ordered_transaction_id_rec(turn_entry_t *const ordered_transacti
 }
 
 static int find_ordered_transaction_id(const turn_map_t *map, const uint8_t *transaction_id) {
-	return find_ordered_transaction_id_rec(map->ordered_transaction_ids, transaction_id, 0,
-	                                       map->transaction_ids_count);
+	int index = find_ordered_transaction_id_rec(map->ordered_transaction_ids, transaction_id, 0,
+	                                            map->transaction_ids_count);
+	if (index < 0) {
+		return map->transaction_ids_count;
+	}
```

Comment:
I don't the this necessary if you return the index instead of -1.

Reply:
done